src/OnlineVotingS.API/Controllers/ComplaintsController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using OnlineVotingS.Application.DTO.PostDTO;
+using OnlineVotingS.Application.DTO.PutDTO;
+using OnlineVotingS.Application.Services.Complaint.Requests.Commands;
+using OnlineVotingS.Application.Services.Complaint.Requests.Queries;
+using OnlineVotingS.Domain.Entities;
+
+namespace OnlineVotingS.API.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class ComplaintsController : Controller
+{
+    private readonly ISender Mediator;
+    public ComplaintsController(ISender _mediator)
+    {
+        Mediator = _mediator;
+    }
+
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<Complaints>> Create([FromBody] ComplaintsPostDTO complaintsPost)
+    {
+        if (complaintsPost == null)
+        {
+            return BadRequest("Complaint data is required.");
+        }
+
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        var command = new CreateComplaintCommand(complaintsPost);
+        var response = await Mediator.Send(command);
+
+        return CreatedAtAction(nameof(GetComplaintsByIdCommand), new { complaintText = complaintsPost.ComplaintText}, response);
+    }
+
+    [HttpDelete]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<bool>> Delete([FromQuery] int ComplaintId)
+    {
+        var command = new DeleteComplaintCommand(ComplaintId);
+        var response = await Mediator.Send(command);
+
+        return response
+                  ? NoContent()
+                  : NotFound($"Complaint with ID {ComplaintId} not found.");
+    }
+
+    [HttpPut]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<Complaints>> Update([FromBody] ComplaintsPutDTO complaintsPut)
+    {
+        if (complaintsPut == null || complaintsPut.ComplaintID <= 0)
+        {
+            return BadRequest("Valid Campaign ID is required.");
+        }
+
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        var command = new UpdateComplaintCommand(complaintsPut);
+        var response = await Mediator.Send(command);
+
+        return response == null
+           ? NotFound($"Complaint with ID {response.ComplaintID} not found.")
```

Comment:
Make userID not null in PostDTO and PurDTO to remove these warnings
U must update like this in dtos: public string UserID { get; set; } = null!;